### PR TITLE
MainWindow: use Adw.NavigationView

### PR DESCRIPTION
**Pull request by danirabbit**
Replace deprecated Leaflet with NavigationView.

Since NavigationView doesn't have a property for disabling navigation, we replace the navigation stack to prevent navigation. There's some tricks with waiting for `shown` so that we still get animations

### CODE REVIEW

**Comment by danirabbit** [comment]
@ryonakano Can you see if this fixes #793 for you? I'm thinking it might be a bug with Leaflet on X11

**Comment by ryonakano** [comment]
> @ryonakano Can you see if this fixes #793 for you? I'm thinking it might be a bug with Leaflet on X11

Tried Testing Mode on X11, but unfortunately not:

expand

[test-screen0.webm](https://github.com/user-attachments/assets/867b4ce3-4ffd-4791-8a68-26ad2dc19ebe)

File: src/MainWindow.vala [hunk 1]
```diff
@@ -128,136 +125,138 @@ public class Installer.MainWindow : Gtk.Window {
...
+        disk_view.cancel.connect (() => navigation_view.pop ());
+        disk_view.next_step.connect (() => load_encrypt_view ());
     }
 
-    private void load_check_view () {
-        if (check_ignored) {
-            return;
-        }
-
+    private void load_check_view (string path) {
```
Comment (ryonakano) [inline src/MainWindow.vala:166]: Is there any reason you chose `string` to separate standard install and custom install? I think we may want to use a enum or bool which are more easier to handle.

File: src/Views/DriversView.vala [hunk 2]
```diff
@@ -75,7 +73,7 @@
         content_area.append (message_box);
 
         var back_button = new Gtk.Button.with_label (_("Back"));
-        back_button.clicked.connect (() => ((Adw.Leaflet) get_parent ()).navigate (BACK));
+        back_button.clicked.connect (() => ((Adw.NavigationView) get_parent ()).pop ());
```
Comment (ryonakano) [inline src/Views/DriversView.vala:76]: Honestly I think the each AbstractInstallerView shouldn't know that its parent is Adw.NavigationView; it should just emit a signal like `previous_step ()`, connect to the signal in MainWindow, and call `Adw.NavigationView.pop ()` in its handler. But that's out of scope of this PR, so leaving just a comment.

**Review by ryonakano — COMMENTED** [review]
Confirmed all pages transit to the expected page and back to the previous page.

Code almost looks good, but leaving some comments.

File: src/Views/DriversView.vala (hunk 2 above)
Comment (danirabbit) [reply src/Views/DriversView.vala]: Made it into an action :)

File: src/MainWindow.vala (hunk 1 above)
Comment (danirabbit) [reply src/MainWindow.vala]: Re-wrote it without the strings!

**Review by ryonakano — APPROVED** [review]
Let's go :rocket:

**Pull request merged**